hadoop-project-dist/pom.xml

```diff
@@ -106,7 +106,7 @@
           <source>${maven.compile.source}</source>
           <charset>${maven.compile.encoding}</charset>
           <reportOutputDirectory>${project.build.directory}/site</reportOutputDirectory>
-          <destDir>${project.build.directory}/api</destDir>
+          <destDir>api</destDir>
```

Comment:
This seems to be a revert of HADOOP-13784? Something to do with javadoc-plugin version? I think the intended path was ``${project.build.directory}/api``, post that if we follow the doc that you shared it is like ``${project.build.directory}/site/api``? Can we not do ``../api`` and maintain the behaviour?

And I think it isn't going there either, it going inside ``site/apidocs`` rather than ``site/api``

```
Generating /Users/ayushsaxena/code/hadoop-code/hadoop/hadoop-hdfs-project/hadoop-hdfs-httpfs/target/site/apidocs/org/apache/hadoop/fs/http/server/FSOperations.FSFileChecksum.html...
```

Reply:
Thanks for the review @ayushtkn. 

> And I think it isn't going there either, it going inside site/apidocs rather than site/api

The `hadoop-hdfs-httpfs` module doesn't inherit from `hadoop-project-dist`. So, its `maven-javadoc-plugin` configuration must be coming from somewhere else that points to `site/apidocs`. However, I couldn't find any relevant reference to `site/apidocs` in the Hadoop codebase (I'm thinking it's a default value that's kicking in here).

> Can we not do ../api and maintain the behaviour? And I think it isn't going there either,

The javadoc is getting generated inside `${project.build.directory}/api` for me when I use `../api` -

```powershell
PS C:\Users\Gautham> Get-ChildItem -Recurse -Filter 'target\api' -Path H: -ErrorAction SilentlyContinue | foreach { $_.FullName } | where { (Get-ChildItem $_).Count -gt 0 }
H:\hadoop-hdfs-project\hadoop-hdfs\target\api
H:\hadoop-hdfs-project\hadoop-hdfs-client\target\api
H:\hadoop-hdfs-project\hadoop-hdfs-rbf\target\api
```